code/datums/controllers/throwing_controls.dm

```diff
@@ -121,10 +123,18 @@ var/global/datum/controller/throwing/throwing_controller = new
 					new /obj/effects/explosion/dangerous(next)
 					end_throwing = FALSE
 					thr.throw_type = THROW_NORMAL
+
+			if (thr.throw_type == THROW_PHASE && thr.get_throw_travelled() > 2)
+				thr.throw_type = THROW_NORMAL
+				thing.event_handler_flags = initial(thing.event_handler_flags)
+
 			if (!thing.Move(next))  // Grayshift: Race condition fix. bump proc calls are delayed past the end of the loop and won't trigger end condition
-				thr.hitAThing = TRUE // of !throwing on their own, so manually checking if Move failed as end condition
-				end_throwing = TRUE
-				break
+				if (thr.throw_type == THROW_PHASE)
```

Comment:
Why try `Move` and then `set_loc` instead of just `set_loc`ing directly?

Reply:
can you take a look at the new approach taken?